File: relationship-manager/followup-orchestrator/src/main/java/com/bank/rm/followup/dto/FollowUpDtos.java
package com.bank.rm.followup.dto;

import com.bank.rm.common.dto.CommunicationChannel;
import com.bank.rm.common.dto.FollowUpFrequency;
import com.bank.rm.common.dto.FollowUpStatus;
import java.time.LocalDateTime;
import java.util.UUID;

public class FollowUpDtos {

    public record CreateScheduleRequest(UUID customerId, FollowUpFrequency frequency,
                                         CommunicationChannel channel) {}

    public record ScheduleResponse(UUID scheduleId, UUID customerId, FollowUpFrequency frequency,
                                    CommunicationChannel channel, LocalDateTime nextFollowUpAt,
                                    boolean active, int totalCompleted) {}

    public record InstanceResponse(UUID instanceId, UUID customerId, LocalDateTime scheduledAt,
                                    FollowUpStatus status, String agenda) {}

    private FollowUpDtos() {}
}

File: relationship-manager/followup-orchestrator/src/main/kotlin/com/bank/rm/followup/dto/FollowUpDtos.kt
package com.bank.rm.followup.dto

import com.bank.rm.common.dto.*
import java.time.LocalDateTime
import java.util.UUID

data class CreateScheduleRequest(
    val frequency: FollowUpFrequency,
    val channel: CommunicationChannel,
    val startDate: LocalDateTime? = null
)

data class ScheduleResponse(
    val id: UUID,
    val customerId: UUID,
    val frequency: FollowUpFrequency,
    val channel: CommunicationChannel,
    val nextFollowUpAt: String,
    val isActive: Boolean,
    val totalCompleted: Int,
    val totalMissed: Int
)

data class FollowUpInstanceResponse(
    val id: UUID,
    val scheduleId: UUID,
    val customerId: UUID,
    val scheduledAt: String,
    val status: FollowUpStatus,
    val agenda: String,
    val conversationId: UUID?
)
